gradle/lib/dependabot/gradle/file_updater.rb

```diff
@@ -69,6 +68,13 @@ def update_buildfiles_for_dependency(buildfiles:, dependency:)
 
           buildfile = files.find { |f| f.name == new_req.fetch(:file) }
 
+          # Exception raised to handle issue that arises when buildfiles function (see this file)
+          # removes the build file that contains the dependency itself. So no build file exists to
+          # update dependency, This behaviour is evident for extremely small number of users
+          # that have added separate repos as sub-modules in parent projects
+
+          raise DependencyFileNotResolvable, "No build file found to update the dependency" if buildfile.nil?
```

Comment:
Also, I really like your wording of this exception message because it describes what happened, but not assuming how we got into this situation... 

The code comment is helpful to explain one way we get into this situation, but there may be more ways... so keeping the actual exception message pretty generic like you did seems best. 👍